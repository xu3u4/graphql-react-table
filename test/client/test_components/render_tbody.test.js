import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import RenderTbody from 'components/view_issues/render_tbody';

jest.mock('data');

describe('Render <RenderTbody>', () => {
  const { headers, issues } = require('data');
  const handleDeleteRow = sinon.stub(),
        selectIssue = sinon.stub();
  let tbody, newIssue = "1";

  beforeEach(() => {
    tbody = shallow(
      <RenderTbody
        columns={ headers }
        rows={ issues }
        onDeleteIssue={ handleDeleteRow }
        onSelectIssue={ selectIssue }
        newIssue={ newIssue }
      />
    );
  });

  it('should render tr, td', () => {
    tbody = renderer.create(
      <RenderTbody
        columns={ headers }
        rows={ issues }
        onDeleteIssue={ handleDeleteRow }
        onSelectIssue={ selectIssue }
        newIssue={ newIssue }
      />
    );
    expect(tbody).toMatchSnapshot();
  });

  it('should have different color if just updated', () => {
    expect(tbody.childAt(newIssue-1).hasClass('highlight')).toBetruthy;
  });

  it('should select issue when double click', () => {
    tbody.find('tr').first().simulate('doubleclick');
    expect(selectIssue.calledOnce).toBeTruthy;
  });
});
