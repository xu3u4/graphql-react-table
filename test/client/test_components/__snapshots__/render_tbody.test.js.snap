// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <RenderTbody> should render tr, td 1`] = `
<tbody>
  <tr
    className={null}
    onDoubleClick={[Function]}
  >
    <td>
      1
    </td>
    <td>
      Open
    </td>
    <td>
      cat1
    </td>
    <td>
      Java
    </td>
    <td>
      Allen
    </td>
    <td>
      1
    </td>
    <td>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>
  <tr
    className={null}
    onDoubleClick={[Function]}
  >
    <td>
      2
    </td>
    <td>
      Close
    </td>
    <td>
      cat1
    </td>
    <td>
      Go crazy
    </td>
    <td>
      Nick
    </td>
    <td>
      2
    </td>
    <td>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>
  <tr
    className={null}
    onDoubleClick={[Function]}
  >
    <td>
      3
    </td>
    <td>
      Close
    </td>
    <td>
      cat3
    </td>
    <td>
      Wandering
    </td>
    <td>
      Jocelyn
    </td>
    <td>
      3
    </td>
    <td>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>
  <tr
    className={null}
    onDoubleClick={[Function]}
  >
    <td>
      4
    </td>
    <td>
      Open
    </td>
    <td>
      cat1
    </td>
    <td>
      Big belly
    </td>
    <td>
      Haku
    </td>
    <td>
      4
    </td>
    <td>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>
</tbody>
`;
